src/VisualStudio/VisualBasic/Impl/ChangeSignature/VisualBasicChangeSignatureLanguageService.vb

```diff
@@ -0,0 +1,51 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.CSharp.Syntax
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.VisualStudio.LanguageServices.Implementation.ChangeSignature
+Imports Microsoft.VisualStudio.LanguageServices.Implementation.IntellisenseControls
+Imports Microsoft.VisualStudio.Text
+Imports Microsoft.VisualStudio.Text.Projection
+Imports Microsoft.VisualStudio.Utilities
+
+Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.ChangeSignature
+    <ExportLanguageService(GetType(IChangeSignatureLanguageService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicChangeSignatureLanguageService
+        Inherits ChangeSignatureLanguageService
+        Public Overrides Async Function CreateViewModelsAsync(rolesCollectionType() As String,
+                                              rolesCollectionName() As String,
+                                              insertPosition As Integer, document As Document,
+                                              documentText As String, contentType As IContentType,
+                                              intellisenseTextBoxViewModelFactory As IntellisenseTextBoxViewModelFactory,
+                                              cancellationToken As CancellationToken) As Task(Of IntellisenseTextBoxViewModel())
+            Dim rolesCollections = {rolesCollectionName, rolesCollectionType}
+
+            ' We insert '[]' so that we're always able to generate the type even if the name field is empty. 
+            ' We insert '~' to avoid Intellisense activation for the name field, since completion for names does not apply to VB.
```

Comment:
Why does this work?

Reply:
(and more importantly, do we have tests?)

Reply:
@allisonchou, could you please reply?

---
In reply to: [367184827](https://github.com/dotnet/roslyn/pull/38748#discussion_r367184827) [](ancestors = 367184827)

Reply:
[https://github.com/dotnet/roslyn/issues/41060](https://github.com/dotnet/roslyn/issues/41060)

---
In reply to: [367185302](https://github.com/dotnet/roslyn/pull/38748#discussion_r367185302) [](ancestors = 367185302)